sktime/forecasting/statsforecast.py

```diff
@@ -552,3 +555,150 @@ def get_test_params(cls, parameter_set="default"):
         params = [{}, {"season_length": 4, "model": "Z"}]
 
         return params
+
+
+class StatsForecastMSTL(_GeneralisedStatsForecastAdapter):
+    """StatsForecast Multiple Seasonal-Trend decomposition using LOESS model.
+
+    This implementation is a wrapper over Nixtla implementation in
+    statsforecast [1]_.
+
+    The MSTL (Multiple Seasonal-Trend decomposition using LOESS) decomposes the time
+    series in multiple seasonalities using LOESS. Then forecasts the trend using
+    a custom non-seasonal model (`trend_forecaster`) and each seasonality using a
+    SeasonalNaive model. MSTL requires the input time series data to be univariate.
+
+    Parameters
+    ----------
+    season_length : Union[int, List[int]]
+        Number of observations per unit of time. For multiple seasonalities use a
+        list.
+    trend_forecaster : estimator, optional, default=StatsForecastAutoETS()
+        Sktime estimator used to make univariate forecasts. Multivariate estimators are
+        not supported.
+
+    References
+    ----------
+    .. [1]
+        https://nixtla.github.io/statsforecast/src/core/models.html#multiple-seasonalities
```

Comment:
```suggestion
        https://nixtla.github.io/statsforecast/src/core/models.html#mstl
```

It's better to link the model directly.